### ISSUE TITLE
Remove all compilation warnings?

### ISSUE THREAD

**Issue by jjerphan** [pola-rs/polars#5659]
### Problem description

Hi!

First, thank you for developing Polars! :handshake: 

Coming from the [Scientific Python](scientific-python.org/) world, I am curious about Polars (see my as-of-now sole interaction https://github.com/pola-rs/polars/issues/2249#issuecomment-1287374875) and want to contribute to the project.

I have followed all the instructions in the ["Contributing to the codebase"](https://github.com/pola-rs/polars/blob/master/CONTRIBUTING.md#contributing-to-the-codebase) regarding getting started with a development environment but noticed a lot a compilations warnings when compiling the project when running `make` command (due to unused imports, to the declaration of unused variables or to the use of deprecated functions).

Would removing (some of) the compilation warnings be a good first issue?

Thank you for your feedback!

**Comment by stinodego** [comment]
It's definitely worth looking into, and I think it could be a good first issue.

Keep in mind that not all of these are easily fixed. Many of these warnings happen because of the way Polars does feature gating. But I think there is definitely room for improvement.

Looking forward to your contribution!

**Comment by jjerphan** [comment]
Thank you for your prompt answer!

> Many of these warnings happen because of the way Polars does feature gating.

Can you explain this in more details?

In particular, I just have opened #5664 and I am wondering whether it could be blocked by someone.

**Comment by ritchie46** [comment]
Yeah, there are hard warning because of feature gating, but the warnings that are given by the tests should be easy. 

It might not be the most fun, but its definitely thankful work. :)

**Comment by jjerphan** [comment]
Let me know if there are others issues or PR I can help you with. :slightly_smiling_face:

**Comment by matthiasbeyer** [comment]
Hi.
When compiling polars (Rust) I do not get any warnings. Also clippy does not report anything. Is this issue done or am I missing something?

**Comment by jjerphan** [comment]
I think it can be closed. The codebase moves fast, and warnings are introduced and removed frequently.

(I let polars' maintainers close this issue.)

**Comment by stinodego** [comment]
I think this has mostly been solved. If there are any specific warnings someone is concerned about, he/she can open a new issue.

**Comment by matthiasbeyer** [comment]
Maybe now is the time to `deny` the most common warnings, so they are not introduced anymore? :laughing:

**Comment by ritchie46** [comment]
> Maybe now is the time to `deny` the most common warnings, so they are not introduced anymore? 😆

They are. We only have warnings that occur on feature flag combinations not ran in our ci. Default features and all features activated don't have warnings on compilation.
